src/nix-instantiate/nix-instantiate.cc

```diff
@@ -31,7 +31,12 @@ void processExpr(EvalState & state, const Strings & attrPaths,
     bool evalOnly, OutputKind output, bool location, Expr * e)
 {
     if (parseOnly) {
-        std::cout << format("%1%\n") % *e;
+        if (output == okJSON) {
+            std::cout << format("%1%\n") % (ExprAsJson *) *e;
```

Comment:
We shouldn't add this functionality to `nix-instantiate` since it's legacy. Better to add a new command like `nix parse`.

Reply:
yes, thanks for remind. also the CLI of `nix-instantiate` is not ideal (`--json`, `--xml`)

Reply:
first draft in 4c35710